### ISSUE TITLE
Suggestion for the French language

### ISSUE THREAD

**Issue by tomschelsen** [DigitalPhonetics/IMS-Toucan#74]
Disclaimer : this might come down to geo/political preferences, but I thought the information is worth sharing.

When listening to the Meta model in the French language, I found some pronunciation/accent surprising (I am a French native speaker from France). Looking at the code, it seems the Meta model is mainly trained on the CSS10_fr dataset. So I went to listen to this dataset and indeed found the same "suprises" I heard before. Information on the internet is scarce, but it seems the speaker might be from Canada. So, although being definitely French in the grammatical sense, the pronunciation is more `fr-ca` than `fr-fr`.

So if by any chance you intended to retrain the Meta model, and are ok to make the French language + French accent sounding more "French from France", I would suggest avoiding CSS10 and using the Synpaflex Corpus ( https://www.ortolang.fr/market/corpora/synpaflex-corpus , I have seen reference to a "FrenchExpressive" corpus in the code, don't know if it is this one) or the SIWIS dataset ( https://datashare.ed.ac.uk/handle/10283/2353 , at least the subparts that are segmented).

Here are the loading functions I made (the Synpaflex one only take the utterances for which the text was normalized by the authors, but an extended version could easily be made) : 

```
import glob
from pathlib import Path
import os

def build_path_to_transcript_dict_synpaflex_norm_subset():
    root = "/data/inputs/speech_natural/fr/synpaflex-corpus/5/v0.1/"
    path_to_transcript = dict()
    for text_path in glob.iglob(os.path.join(root, "**/*_norm.txt"), recursive=True):
        with open(text_path, "r", encoding="utf8") as file:
            norm_transcript = file.read()
        path_obj = Path(text_path)
        wav_path = str((path_obj.parent.parent / path_obj.name[:-9]).with_suffix(".wav"))
        if Path(wav_path).exists():
            path_to_transcript[wav_path] = norm_transcript
    return path_to_transcript

def build_path_to_transcript_dict_siwis_subset():
    root = "/data/inputs/speech_natural/fr/SiwisFrenchSpeechSynthesisDatabase/"
    # part4 and part5 are not segmented
    sub_dirs = ["part1", "part2", "part3"]
    path_to_transcript = dict()
    for sd in sub_dirs:
        for text_path in glob.iglob(os.path.join(root, "text", sd, "*.txt")):
            with open(text_path, "r", encoding="utf8") as file:
                norm_transcript = file.read()
            path_obj = Path(text_path)
            wav_path = str((path_obj.parent.parent.parent / "wavs" / sd / path_obj.stem).with_suffix(".wav"))
            if Path(wav_path).exists():
                path_to_transcript[wav_path] = norm_transcript
    return path_to_transcript
```

**Comment by Flux9665** [comment]
Thank you very much for sharing! As someone who doesn't speak French, I had no idea that the CSS10-French is not the European French. As you said, there is little documentation about this data. I will add the new french data to the next version of the toolkit which comes with an entirely new architecture, so I will also definitely re-train the large multilingual model. In general, dialects and accents are difficult in the TTS case, because those things are often poorly annotated. The multilingual librispeech for French probably also includes a lot of French that is not the variant spoken in France.

**Comment by tomschelsen** [comment]
Thanks a lot ! I can leave the issue open as a reminder or close it to tidy things up, as you prefer.

**Comment by Flux9665** [comment]
Let's leave it open as reminder, I will close it once the release that supports the new datasets is out.

**Comment by Flux9665** [comment]
The data handling is now in the experimental branch that will become the new main branch once it is finished: 

https://github.com/DigitalPhonetics/IMS-Toucan/blob/9a4e10330cd2253a1264ee997465dbd3c9cc1240/TrainingInterfaces/TrainingPipelines/PortaSpeech_MetaCheckpoint.py#L157

Also, since the Blizzard Challenge 2023 is based on French data, I included those datasets as well. Blizzard Challenge Datasets are usually very high quality. So hopefully the next version will speak much better european French :)

**Comment by Flux9665** [comment]
Today's release makes those datasets the new default for the French part of the multilingual setup together with the previously used Multilingual Librispeech French dataset. The model included in the new release is trained on the French-French datasets. If it still sounds like Canadian-French, it must be due to the Multilingual Librispeech French dataset, which is still the largest amount of French data in the pretraining. A demo that requires no setup is running here: https://huggingface.co/spaces/Flux9665/ThisSpeakerDoesNotExist

Can you please let me know if it still sounds Canadian to you as native speaker? In that case I'll probably separate the MLS dataset out and give it its own language ID, I did the same with European Portuguese and Brazilian Portuguese.

**Comment by tomschelsen** [comment]
Congrats on the new release :) It does definitely sounds better accent-wise.

I heard some things that were incorrect especially on numbers, but thanks to the new visualisation you put in place in the demo, I realised the phonemizer was to blame for that. I guess I'll have to open issues on another repository ;)

So we can close this one.

I'm waiting on some data but then will try to fine-tune with the new architecture and keep you advised if any performance / usability issues arise.

Thanks again for your work.

**Comment by Ca-ressemble-a-du-fake** [comment]
I remembered this discussion while I was struggling to make Meta voice pronounce "ɛ̃" (like in "dessin", or "un humain") in French from France. That's why I added a comment here.

If I generate "pas besoin de vous faire un dessin" it sounds like "pas besoin de vous faire un dessai" so the last "in" is pronounced like "ẽ" instead of "ɛ̃" although the first one (in "besoin") is OK.

https://user-images.githubusercontent.com/91517923/225521496-657699d1-7039-4c1c-b719-74194aacacc7.mp4

Nearly all ending "ɛ̃" are mispronounced in sentences. But if I give it "le matin, le patin, le gratin," the pronunciation is OK.

I tried all variant of "ɛ̃" (dessin, dessein, dessain) but the model keeps saying it like "ẽ". I am not sure about "humain" and "chemin" to be pronounced as fr-ca or fr-fr.

@tomschelsen you mentionned a phonemizer problem but here it looks like it does the right job :
pa bəzw**ˈɛ** də vu fˈɛʁ œ dɛs**ˈɛ**.~#

Is there a way to make the Meta model always pronounce "ɛ̃" consistently as fr-fr or it is "soldered" onto the model for ever and cannot be corrected :wink ?
